src/sqlite3/connection.cr

```diff
@@ -1,9 +1,19 @@
 class SQLite3::Connection < DB::Connection
+  @@regexp : (LibSQLite3::SQLite3Context, Int32, LibSQLite3::SQLite3Value*) -> = ->(context : LibSQLite3::SQLite3Context, argc : Int32, argv : LibSQLite3::SQLite3Value*) {
```

Comment:
I think the declaration is not required.

```suggestion
  @@regexp = ->(context : LibSQLite3::SQLite3Context, argc : Int32, argv : LibSQLite3::SQLite3Value*) {
```

You should be able to use also a block syntax if the `FuncCallback` is declared.

```cr
@@regexp = LibSQLite3::FuncCallback.new do |context, argc, argv|
  argv = Slice.new(argv, sizeof(Void*))
  pattern = LibSQLite3.value_text(argv[0])
  text = LibSQLite3.value_text(argv[1])
  LibSQLite3.result_int(context, Regex.new(String.new(pattern)).matches?(String.new(text)).to_unsafe)
end
```

Since it's a constant, it can be stored as such instead of a class var `SQLite3::REGEXP_FN` maybe?